### PR TITLE
Interpreter: fix fiber's resumable property

### PR DESCRIPTION
**Pull request by ysbaddaden**
The `resumable` property of fiber contexts was wrong in interpreted code: it was never reset to `0`.

This patch lets the interpreted fiber read the resumable property of the interpreter's fiber (the real one) though a new primitive. The interpreter is now responsible from reporting the value properly. This avoids having to manipulate `resumable` in the interpreted code, and allows to act on the actual state.

*Alternative*: the interpreter could set the `resumable` properties of the fibers in the `interpreter_swapcontext` primitive, it would be fake but probably not critical (the interpreter's scheduler would still act on the real property), but we don't have references back into the interpreted fibers (we could add it when interpreted) + I don't know how the interpreter updates values into an interpreted type (I could learn).

### CODE REVIEW

File: src/fiber/context.cr [hunk 1]
```diff
@@ -7,10 +7,26 @@ class Fiber
...
-    property resumable : LibC::Long
+
+    {% if flag?(:interpreted) %}
+      # In interpreted mode the interpreted fiber's will be backed by a real
+      # fiber run by the interpreter. The fiber context is thus fake.
+      #
+      # The `stack_top` property is actually a pointer to the real Fiber
+      # running in the interpreter.
+      #
+      # Th `resumable` proprety is also delegated to the real fiber. Only the
```
Comment (beta-ziliani) [inline src/fiber/context.cr:18]:
```suggestion
      # The `resumable` proprety is also delegated to the real fiber. Only the
```

**Review by beta-ziliani — COMMENTED** [review]
Is it possible to have a test that should break in the current HEAD but work with this patch?

**Review by straight-shoota — APPROVED** [review]

File: spec/std/fiber_spec.cr [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  it "#resumable?" do
+    done = false
+    resumable = nil
+
+    fiber = spawn do
+      resumable = Fiber.current.resumable?
+      done = true
+    end
+
+    fiber.resumable?.should eq(true)
```
Comment (Sija) [inline spec/std/fiber_spec.cr:13]:
```suggestion
    fiber.resumable?.should be_true
```

File: spec/std/fiber_spec.cr [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    until done
+      Fiber.yield
+    end
+
+    resumable.should eq(false)
```
Comment (Sija) [inline spec/std/fiber_spec.cr:19]:
```suggestion
    resumable.should be_false
```

**Review by Sija — COMMENTED** [review]
Nitpick

**Comment by ysbaddaden** [comment]
@beta-ziliani I just added a test. It fails with the interpreter of crystal 1.11 but succeeds otherwise (this patch, or when compiled).

File: src/fiber/context.cr (hunk 1 above)
Comment (ysbaddaden) [inline src/fiber/context.cr:18]:
```suggestion
      # The `resumable` property is also delegated to the real fiber. Only the
```

**Review by beta-ziliani — APPROVED** [review]

**Pull request merged**